### PR TITLE
Add deprecation notices to deprecated plugins

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: internal/signalfx-agent/pkg/monitors/collectd/monitorcore.go [hunk 1]
```diff
@@ -63,9 +63,17 @@ func (mc *MonitorCore) collectdInstance() *Manager {
 	return MainInstance()
 }
 
+type ConfigurationOption func(mc *MonitorCore)
+
+func WithDeprecationWarningLog(message string) ConfigurationOption {
+	return func(mc *MonitorCore) {
+		mc.logger.Warn(message)
+	}
+}
```
[the remark this answers is not in the document]
Comment (atoulme) [reply internal/signalfx-agent/pkg/monitors/collectd/monitorcore.go]: ok done

**Review by rmfitzpatrick — APPROVED** [review]

**Review by jvoravong — APPROVED** [review]

**Comment by atoulme** [comment]
`collectd/elasticsearch` uses a different method, and has a warning already:
```
m.Logger().Warn("The collectd/elasticsearch monitor is deprecated in favor of the elasticsearch monitor.")
```
https://github.com/signalfx/splunk-otel-collector/blob/03550440ac58cbca3658a6f63fab845e25fe2a19/internal/signalfx-agent/pkg/monitors/collectd/elasticsearch/elasticsearch.go#L77

File: internal/signalfx-agent/pkg/monitors/collectd/df/metadata.yaml [hunk 2]
```diff
@@ -8,6 +8,8 @@ monitors:
     namespace that the agent is running in for this monitor to be able to
     collect statistics about that filesystem.  This is mostly an issue when
     running the agent in a container.
+    
+    **This plugin is deprecated in favor of the `filesystems` monitor.**
```
Comment (aurbiztondo-splunk) [inline internal/signalfx-agent/pkg/monitors/collectd/df/metadata.yaml:12]: @atoulme - should we recommend the host metrics receiver instead? That's what we recommend in docs for the filesystems monitor.

File: internal/signalfx-agent/pkg/monitors/collectd/disk/disk.go [hunk 3]
```diff
@@ -38,5 +38,5 @@ type Monitor struct {
 
 // Configure configures and runs the plugin in collectd
 func (m *Monitor) Configure(conf *Config) error {
-	return m.SetConfigurationAndRun(conf)
+	return m.SetConfigurationAndRun(conf, collectd.WithDeprecationWarningLog("disk-io"))
```
Comment (aurbiztondo-splunk) [inline internal/signalfx-agent/pkg/monitors/collectd/disk/disk.go:41]: @atoulme - can we provide an alternative? Would the host metrics receiver do the deed, for instance?

File: internal/signalfx-agent/pkg/monitors/collectd/df/metadata.yaml (hunk 2 above)
Comment (atoulme) [reply internal/signalfx-agent/pkg/monitors/collectd/df/metadata.yaml:12]: I'm just following the advice that was already given back when, we can look at deprecating `filesystems` separately next if the hostmetricsreceiver is ready.

Comment (aurbiztondo-splunk) [reply internal/signalfx-agent/pkg/monitors/collectd/df/metadata.yaml:12]: Ack!

**Pull request merged**